File: python/1-Two-Sum.py
# Space O(n)
# Time O(n)
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        indicesMap = {}
        for i, num in enumerate(nums):
            if target - num in indicesMap:
                return i, indicesMap[target - num]
            indicesMap[num] = i

File: c#/1-Two-Sum.cs
// Time  O(n) where n is nums.Length
// Space O(n) if every elem is unique
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        Dictionary<int, int> valToIdx = new();

        for (int i = 0; i < nums.Length; i++)
        {
            int comp = target - nums[i];
            if (valToIdx.ContainsKey(comp))
            {
                return new int[] { valToIdx[comp], i };
            }
            valToIdx[nums[i]] = i;
        }
        return null;
    }
}
